File: resources/asteroids/scripts/Play.js
// The SceneDirectorScript while playing the game - there is a different SceneDirectorScript for the menus (Menu.js)
Play = new Class({

    Extends: SceneDirectorScript,
    
    init: function()
    {
        this.rocks = 0; // count the rocks on the screen, when it goes down to zero, next level!
        this.endTimer = null;
        this.ship = null; // Set by Ship's onBirth.
    },
    
    onActivate: function() {
        game.loadScene("foreground", true);
    },

    onKeyDown: function(ke)
    {
        // Escape key takes us back to the menu.
        if (ke.symbol == ke.ESCAPE) {
            game.startScene("menu");
            return true; // Return true to indicate that the key has been processed.
        }
        // Play again if dead an return pressed.
        if ((ke.symbol == ke.RETURN) && (directorScript.lives == 0)) {
        	directorScript.startGame();
        }
        return false;
    },
    
    addRocks:function( diff )
    {
    	this.rocks += diff;
    	if (this.rocks == 0) {
			if ((this.ship != null) && (this.ship.actor.isAlive())) {
    			this.endTimer= new itchy.extras.Timer.createTimerSeconds(3);
    			this.ship.warp();
    	    }
    	}
    },
    
    tick: function()
    {
    	if (this.endTimer && this.endTimer.isFinished()) {
    		var nextScene = 1 + parseInt(game.getSceneName());
    		if (game.hasScene( nextScene )) {
    			game.startScene( nextScene );
    		} else {
    			game.startScene("completed");
    		}
    	}
    }
    
});

File: resources/Destroy-Debris/scripts/play.py
from common import *

properties = ArrayList()
properties.add( StringProperty("nextScene").hint("The name of the next scene") )

game = Itchy.getGame()
director = game.getDirector()

class Play(PlainSceneDirector) :

    def __init__(self) :
        self.rocks = 0 # count the rocks on the screen, when it goes down to zero, next level!
        self.endTimer = None
        self.ship = None # Set by Ship's onBirth.
        self.nextScene = "completed"

        self.inputExit = Input.find("exit")
        self.inputPause = Input.find("pause")
        self.inputRestart = Input.find("restart")
        self.inputContinue = Input.find("continue")


    def onLoaded(self) :
        print "Loading glass"
        game.loadScene("glass", True)


    def onKeyDown(self, event) :
    
        # Escape key takes us back to the menu.
        if self.inputExit.matches(event) :
            game.startScene("menu")
            event.stopPropagation()

        if director.lives == 0 :
        
            # Play again if dead an return pressed.
            if self.inputRestart.matches(event) :
                director.startGame()

            if self.inputContinue.matches(event) :
                director.startGame(game.getSceneName())
    
        if self.inputPause.matches(event) :
            game.pause.togglePause()
    
    
    def addRocks(self, diff) :

        self.rocks += diff
        if self.rocks == 0 :
            if self.ship != None and self.ship.getActor().isAlive() :
                self.endTimer = Timer.createTimerSeconds(3)
                self.ship.warp()


    def tick(self) :
    
        if self.endTimer and self.endTimer.isFinished() :
            if game.hasScene( self.nextScene ) :
                game.startScene( self.nextScene )
            else :
                game.startScene("completed")   
    
    
    def getCollisionStrategy(self,actor) :
        return director.collisionStrategy
    
    
    # Boiler plate code - no need to change this
    def getProperties(self):
        return properties

    # Boiler plate code - no need to change this
    def getClassName(self):
        return ClassName( SceneDirector, self.__module__ + ".py" )


